runtime/src/snapshot_utils.rs

```diff
@@ -377,8 +382,9 @@ pub fn archive_snapshot_package(
 
     // Write version file
     {
-        let mut f = fs::File::create(staging_version_file)
-            .map_err(|e| SnapshotError::IoWithSource(e, "create version file"))?;
+        let mut f = fs::File::create(&staging_version_file).map_err(|e| {
+            SnapshotError::IoWithSourceAndFile(e, "create version file", staging_version_file)
+        })?;
         f.write_all(snapshot_package.snapshot_version.as_str().as_bytes())
             .map_err(|e| SnapshotError::IoWithSource(e, "write version file"))?;
```

Comment:
Should this IoWithSource be updated to IoWithSourceAndFile?

Wondering if there is any reason to have IoWithSource exist without providing the path info.